chore: 3.8 breaking changes (#7818)
Kong/docs.konghq.com: app/_src/gateway/breaking-changes/38x.md

When upgrading, set the environment variable `DEBIAN_FRONTEND=noninteractive` on Debian/Ubuntu to avoid any interactive prompts and enable fully automatic upgrades.

### `log_statistics` defaults to `false` for AI Proxy
A configuration validation was added to the [AI Proxy plugin](/hub/kong-inc/ai-proxy/) to prevent users from enabling `log_statistics` for
providers that don't support statistics. In addition, the default of `log_statistics` was changed from
`true` to `false`, and a database migration is added as well for disabling `log_statistics` if it
has already been enabled upon unsupported providers.

### Deprecations

#### `sentinel_addresses` and `cluster_addresses` for Redis

The following plugins switched `cluster_addresses` to `cluster_nodes` and `sentinel_addresses` to `sentinel_nodes` for Redis configuration:

* [AI Rate Limiting Advanced](/hub/kong-inc/ai-rate-limiting-advanced/)
* [GraphQL Proxy Caching Advanced](/hub/kong-inc/graphql-proxy-cache-advanced/)
* [GraphQL Rate Limiting Advanced](/hub/kong-inc/graphql-rate-limiting-advanced/)
* [Proxy Caching Advanced](/hub/kong-inc/proxy-cache-advanced/)

> * [Rate Limiting Advanced](/hub/kong-inc/rate-limiting-advanced/)

Author: Hi @jschmid1 ! I'm writing a doc that compiles the 3.8 breaking changes and I had a question about the `sentinel_addresses` and `cluster_addresses` for Redis feature changes you made. Are these converted automatically with 3.8 or does the user have to manually migrate or change anything? Thanks!

## Breaking changes and deprecations

### New DNS client

> Starting 3.8.x, a new DNS client library was implemented and added to {{site.base_gateway}}. The new DNS client introduces and new standardized way to configure a service, and helps improve performance. It is disabled by default, but will be enabled by default in a future release. For more information, see [Migrate to the new DNS client](/gateway/migrate-to-new-dns-client/).

Reviewer: This isn't technically a breaking change yet, but it will be if they turn it on. Not sure that it belongs here.

### `kong.logrotate` configuration file no longer overwritten during upgrade
The `kong.logrotate` configuration file will no longer be overwritten during upgrade.
When upgrading, set the environment variable `DEBIAN_FRONTEND=noninteractive` on Debian/Ubuntu to avoid any interactive prompts and enable fully automatic upgrades.

### `log_statistics` defaults to `false` for AI Proxy
A configuration validation was added to the [AI Proxy plugin](/hub/kong-inc/ai-proxy/) to prevent users from enabling `log_statistics` for
providers that don't support statistics. In addition, the default of `log_statistics` was changed from
`true` to `false`, and a database migration is added as well for disabling `log_statistics` if it
has already been enabled upon unsupported providers.

### Custom plugins that used shared Redis config

> In 3.8.0.0, Kong has changed and refactored the shared Redis configuration that previously was imported by `require "kong.enterprise_edition.redis"`. If you created a custom plugin that is using this shared configuration or if you have a forked versionof a plugin, like `rate-limiting-advanced`, then you might need to do additional steps before you can upgrade to the new version of this Redis config.

Reviewer: ```suggestion
In 3.8.0.0, Kong has changed and refactored the shared Redis configuration that previously was imported by `require "kong.enterprise_edition.redis"`. If you created a custom plugin that is using this shared configuration or if you have a forked version of a plugin, like `rate-limiting-advanced`, then you might need to do additional steps before you can upgrade to the new version of this Redis config.
```

> Out-of-the-box the custom plugins should still work since the old shared configuration was left in place. The new config adds the `cluster_max_redirections` option for Redis Cluster, and the `cluster_nodes` format and

Reviewer: ```suggestion
Out of the box, custom plugins should still work since the old shared configuration is still available. The new config adds the `cluster_max_redirections` option for Redis Cluster, and the `cluster_nodes` format and
```
